app/models/robots/robot.rb

```diff
@@ -74,18 +75,35 @@ def bed_plates=(bed_plates)
     private
 
     def valid_robot
-      if verify_robot? && beds.values.first.plate.present?
-        if beds.values.first.plate.custom_metadatum_collection.nil?
-          error_messages << 'Your plate is not on the right robot'
-          return false
-        elsif beds.values.first.plate.custom_metadatum_collection.metadata['created_with_robot'] != robot_barcode
-          error_messages << 'Your plate is not on the right robot'
-          return false
-        end
+      return false unless robot_present_if_required
+
+      return true unless verify_robot? && beds.values.first.plate.present?
```

Comment:
Shouldn't this be return true if rather than unless?

Reply:
No, it's the equivalent of the `if verify_robot? && beds.values.first.plate.present?` that was there previously.
i.e. You only want to bother doing the validation to check if the robot is the same as before IF the verify_robot flag is set and there is a plate barcode scanned in

I'll change it back to how it was before as think that's more readable

Reply:
Ah I see :) 
I read verify_robot as a method you'd always want to run rather than a flag, so returning true if the robot was invalid seemed wrong. It's fine as the guard clause.